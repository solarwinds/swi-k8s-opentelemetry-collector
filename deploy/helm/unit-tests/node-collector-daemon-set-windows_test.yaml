# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for node-collector-daemon-set-windows
templates:
  - node-collector-daemon-set-windows.yaml
  - node-collector-config-map-windows.yaml
  - common-env-config-map.yaml
chart:
  appVersion: 1.0.0
tests:
  - it: DaemonSet spec for windows nodes should match snapshot when using default values
    template: node-collector-daemon-set-windows.yaml
    asserts:
      - matchSnapshot:
          path: spec.template.spec
  - it: Image should be correct in default state
    template: node-collector-daemon-set-windows.yaml
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: solarwinds/swi-opentelemetry-collector:1.0.0
  - it: Image should be correct when overridden tag
    template: node-collector-daemon-set-windows.yaml
    set:
      otel.windows.image.tag: "beta1"
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: solarwinds/swi-opentelemetry-collector:beta1
  - it: Image should be correct when overridden by azure
    template: node-collector-daemon-set-windows.yaml
    set:
      aks: true
      global.azure.images.image_windows.digest: "abcd"
      global.azure.images.image_windows.image: "swi-opentelemetry-collector:v1.2.3"
      global.azure.images.image_windows.registry: "azurek8s.azure.io/marketplaceimages"
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: azurek8s.azure.io/marketplaceimages/swi-opentelemetry-collector:v1.2.3@abcd
  - it: DaemonSet spec for windows nodes should match snapshot when overriding cluster ID
    template: node-collector-daemon-set-windows.yaml
    set:
      cluster.uid: customUid
    asserts:
      - matchSnapshot:
          path: spec.template.spec
  - it: SWI endpoint check should not be included when disabled
    template: node-collector-daemon-set-windows.yaml
    set:
      otel.swi_endpoint_check.enabled: false
    asserts:
    - notExists:
        path: spec.template.spec.initContainers
  - it: SWI endpoint check should be included in default state
    template: node-collector-daemon-set-windows.yaml
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].name
          value: otel-endpoint-check
  - it: SWI endpoint check should not be included when disabled by deprecated setting
    template: node-collector-daemon-set-windows.yaml
    set:
      otel.metrics.swi_endpoint_check: false
    asserts:
    - notExists:
        path: spec.template.spec.initContainers
  - it: Image otel-endpoint-check should be correct in default state
    template: node-collector-daemon-set-windows.yaml
    set:
      otel.swi_endpoint_check.enabled: true
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: solarwinds/swi-opentelemetry-collector:1.0.0
  - it: Image otel-endpoint-check should be correct when overridden repository
    template: node-collector-daemon-set-windows.yaml
    set:
      otel.swi_endpoint_check.enabled: true
      otel.windows.image.repository: "swi-opentelemetry-collector"
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: swi-opentelemetry-collector:1.0.0
  - it: Image otel-endpoint-check should be correct when overridden tag
    template: node-collector-daemon-set-windows.yaml
    set:
      otel.swi_endpoint_check.enabled: true
      otel.windows.image.tag: "beta1"
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: solarwinds/swi-opentelemetry-collector:beta1
  - it: Image otel-endpoint-check should be correct when overridden by azure
    template: node-collector-daemon-set-windows.yaml
    set:
      otel.swi_endpoint_check.enabled: true
      aks: true
      global.azure.images.image_windows.digest: "abcd"
      global.azure.images.image_windows.image: "swi-opentelemetry-collector:v1.2.3"
      global.azure.images.image_windows.registry: "azurek8s.azure.io/marketplaceimages"
    asserts:
    - equal:
          path: spec.template.spec.initContainers[0].image
          value: azurek8s.azure.io/marketplaceimages/swi-opentelemetry-collector:v1.2.3@abcd
