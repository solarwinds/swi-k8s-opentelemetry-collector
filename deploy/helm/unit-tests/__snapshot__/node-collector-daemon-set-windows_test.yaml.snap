DaemonSet spec for windows nodes should match snapshot when overriding cluster ID:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: eks.amazonaws.com/compute-type
                  operator: NotIn
                  values:
                    - fargate
    containers:
      - command:
          - c:\wrapper.exe
          - c:\swi-otelcol.exe
          - --config=c:\conf\relay.yaml
          - --feature-gates=filelog.container.removeOriginalTimeField
        env:
          - name: CHECKPOINT_DIR
            value: c:/var/lib/swo/checkpoints
          - name: POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
          - name: GOMEMLIMIT
            valueFrom:
              resourceFieldRef:
                resource: limits.memory
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:1.0.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
        resources:
          limits:
            memory: 1Gi
          requests:
            memory: 50Mi
        securityContext:
          windowsOptions:
            runAsUserName: ContainerAdministrator
        volumeMounts:
          - mountPath: c:\var\log\pods
            name: varlogpods
            readOnly: true
          - mountPath: c:\var\log\containers
            name: varlogcontainers
            readOnly: true
          - mountPath: c:\var\lib\docker\containers
            name: varlibdockercontainers
            readOnly: true
          - mountPath: c:\conf
            name: opentelemetry-collector-configmap
            readOnly: true
          - mountPath: c:/var/lib/swo/checkpoints
            name: logcheckpoints
    initContainers:
      - command:
          - c:\swi-otelcol.exe
          - test-connection
          - --endpoint=$(OTEL_ENVOY_ADDRESS)
          - --apitoken=$(SOLARWINDS_API_TOKEN)
          - --clusteruid=$(CLUSTER_UID)
          - --insecure=$(OTEL_ENVOY_ADDRESS_TLS_INSECURE)
        env:
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:1.0.0
        imagePullPolicy: IfNotPresent
        name: otel-endpoint-check
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: windows
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    terminationGracePeriodSeconds: 600
    tolerations:
      - effect: NoSchedule
        operator: Exists
    volumes:
      - hostPath:
          path: c:\var\log\pods
        name: varlogpods
      - hostPath:
          path: c:\var\log\containers
        name: varlogcontainers
      - hostPath:
          path: c:\var\lib\docker\containers
        name: varlibdockercontainers
      - hostPath:
          path: c:/var/lib/swo/checkpoints/customUid
          type: DirectoryOrCreate
        name: logcheckpoints
      - configMap:
          items:
            - key: logs.config
              path: relay.yaml
          name: RELEASE-NAME-swo-k8s-collector-node-collector-config-windows
        name: opentelemetry-collector-configmap
DaemonSet spec for windows nodes should match snapshot when using default values:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: eks.amazonaws.com/compute-type
                  operator: NotIn
                  values:
                    - fargate
    containers:
      - command:
          - c:\wrapper.exe
          - c:\swi-otelcol.exe
          - --config=c:\conf\relay.yaml
          - --feature-gates=filelog.container.removeOriginalTimeField
        env:
          - name: CHECKPOINT_DIR
            value: c:/var/lib/swo/checkpoints
          - name: POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
          - name: GOMEMLIMIT
            valueFrom:
              resourceFieldRef:
                resource: limits.memory
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:1.0.0
        imagePullPolicy: IfNotPresent
        livenessProbe:
          httpGet:
            path: /
            port: 13133
        name: swi-opentelemetry-collector
        ports:
          - containerPort: 8888
            name: http
            protocol: TCP
        readinessProbe:
          httpGet:
            path: /
            port: 13133
        resources:
          limits:
            memory: 1Gi
          requests:
            memory: 50Mi
        securityContext:
          windowsOptions:
            runAsUserName: ContainerAdministrator
        volumeMounts:
          - mountPath: c:\var\log\pods
            name: varlogpods
            readOnly: true
          - mountPath: c:\var\log\containers
            name: varlogcontainers
            readOnly: true
          - mountPath: c:\var\lib\docker\containers
            name: varlibdockercontainers
            readOnly: true
          - mountPath: c:\conf
            name: opentelemetry-collector-configmap
            readOnly: true
          - mountPath: c:/var/lib/swo/checkpoints
            name: logcheckpoints
    initContainers:
      - command:
          - c:\swi-otelcol.exe
          - test-connection
          - --endpoint=$(OTEL_ENVOY_ADDRESS)
          - --apitoken=$(SOLARWINDS_API_TOKEN)
          - --clusteruid=$(CLUSTER_UID)
          - --insecure=$(OTEL_ENVOY_ADDRESS_TLS_INSECURE)
        env:
          - name: SOLARWINDS_API_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
                optional: true
        envFrom:
          - configMapRef:
              name: RELEASE-NAME-swo-k8s-collector-common-env
        image: solarwinds/swi-opentelemetry-collector:1.0.0
        imagePullPolicy: IfNotPresent
        name: otel-endpoint-check
    nodeSelector:
      kubernetes.io/arch: amd64
      kubernetes.io/os: windows
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    terminationGracePeriodSeconds: 600
    tolerations:
      - effect: NoSchedule
        operator: Exists
    volumes:
      - hostPath:
          path: c:\var\log\pods
        name: varlogpods
      - hostPath:
          path: c:\var\log\containers
        name: varlogcontainers
      - hostPath:
          path: c:\var\lib\docker\containers
        name: varlibdockercontainers
      - hostPath:
          path: c:/var/lib/swo/checkpoints/CLUSTER_NAME
          type: DirectoryOrCreate
        name: logcheckpoints
      - configMap:
          items:
            - key: logs.config
              path: relay.yaml
          name: RELEASE-NAME-swo-k8s-collector-node-collector-config-windows
        name: opentelemetry-collector-configmap
